modules/markup/renderer.go

```diff
@@ -163,6 +166,18 @@ type nopCloser struct {
 
 func (nopCloser) Close() error { return nil }
 
+func renderIFrame(ctx *RenderContext, renderer Renderer, input io.Reader, output io.Writer) error {
+	_, err := io.WriteString(output, fmt.Sprintf(`<iframe src="%s%s/%s/render/%s/%s" name="ifd"
+onload="this.height=ifd.document.body.scrollHeight" width="100%%" scrolling="no" frameborder="0"/>`,
+		setting.AppURL,
+		ctx.Metas["user"],
+		ctx.Metas["repo"],
+		ctx.Metas["BranchNameSubURL"],
+		url.PathEscape(ctx.RelativePath),
+	))
```

Comment:
AppURL => AppSubURL with escaping
user => escaping
repo => escaping
BranchNameSubURL: it should have been escaped.
name="ifd" ... ifd.document: not sure it follows the HTML/JS standard to use the `name` as a variable name.